download.go

```diff
@@ -0,0 +1,23 @@
+package attache
+
+import (
+	"io"
+	"net/http"
+	"path/filepath"
+	"strings"
+)
+
+func (s Server) handleDownload(w http.ResponseWriter, r *http.Request) {
+	fullpath := strings.TrimPrefix(r.URL.RequestURI(), s.GetPrefixPath)
+	objectKey := filepath.Base(fullpath)
+	stream, err := s.Storage.Download(r.Context(), objectKey)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if stream == nil {
+		http.Error(w, fullpath, http.StatusNotFound)
+		return
+	}
+	io.Copy(w, stream)
```

Comment:
NOTE that we didn't set `Content-Type` response header yet

1. We can sniff again and set the `Content-Type` (oh noes, gotta read 512 bytes and juggle rewinding the `io.Reader`...)
2. When we store the file into backend (e.g. s3, google cloud storage, ...) we should've put the sniffed content type as metadata (e.g. see [PutObjectInput.Metadata](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#PutObjectInput). So that when we perform download, we can resupply the `Content-Type` without re-sniffing?